programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -592,21 +607,29 @@ static void on_event(const base_event *ev) {
             ziti_service **zs;
             int idx = 0;
             if (svc_ev->removed_services != NULL) {
-                svc_event.RemovedServices = calloc(sizeof(svc_ev->removed_services), sizeof(struct tunnel_service_s));
+                int svc_array_length = 0;
                 for (zs = svc_ev->removed_services; *zs != NULL; zs++) {
-                    tunnel_service *svc = find_tunnel_service(id, (*zs)->id);
+                    svc_array_length++;
+                }
+                svc_event.RemovedServices = calloc(svc_array_length + 1, sizeof(struct tunnel_service_s));
+                for (int svc_idx = 0; svc_ev->removed_services[svc_idx]; svc_idx++) {
+                    tunnel_service *svc = find_tunnel_service(id, svc_ev->removed_services[svc_idx]->id);
                     if (svc == NULL) {
-                        svc = get_tunnel_service(id, *zs);
+                        svc = get_tunnel_service(id, svc_ev->removed_services[svc_idx]);
                     }
                     svc_event.RemovedServices[idx++] = svc;
```

Comment:
should this use `idx` or `svc_idx`? It seems like `idx` is not needed, at least in this loop?

Reply:
fixed the index